tests/legacy-cli/e2e/tests/commands/add/npm-env-vars.ts

```diff
@@ -0,0 +1,35 @@
+import { expectFileNotToExist, expectFileToExist, writeFile } from '../../../utils/fs';
+import { getActivePackageManager, setRegistry } from '../../../utils/packages';
+import { git, ng } from '../../../utils/process';
+import {
+  createNpmConfigForAuthentication,
+  resetNpmEnvVars,
+  setNpmEnvVarsForAuthentication,
+} from '../../../utils/registry';
+
+export default async function () {
+  const packageManager = getActivePackageManager();
+
+  if (packageManager === 'npm') {
```

Comment:
We should test these with Yarn as well. Yarn does respect NPM_CONFIG.

See: https://classic.yarnpkg.com/en/docs/envvars/

Reply:
I had added this because I was getting Yarn-specific test failures along the way.  I'll try and remove this to see if my test is in a better place now.

Reply:
The Yarn -specific test failed again.  I'll dig in to see if I can get more detail as to why.

Reply:
It appears that Yarn has a special variable for the registry `YARN_REGISTRY`. Let me at this a bit.

Reply:
This will require some additional changes, including infra changes.

Let me follow up in a separate PR

Reply:
Will do.